File: answers/leetcode/Isomorphic Strings/Isomorphic Strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        if len(s) != len(t):
            return False
        char_map = {}
        count_map = {}
        for i in range(len(s)):
            if s[i] not in char_map:
                char_map[s[i]] = t[i]
                if t[i] not in count_map:
                    count_map[t[i]] = 1
                else:
                    return False
            elif char_map[s[i]] != t[i]:
                return False
        return True

File: answers/leetcode/Isomorphic Strings/Isomorphic Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if (null == s && null == t) {
            return true;
        } else if (null == s || null == t || s.length() != t.length()) {
            return false;
        }

        Map<Character, Character> sToT = new HashMap<Character, Character>();
        Map<Character, Character> tToS = new HashMap<Character, Character>();
        for (int i = 0; i < s.length(); i++) {
            if (sToT.containsKey(s.charAt(i))) {
                if (sToT.get(s.charAt(i)) != t.charAt(i)) {
                    return false;
                }
            } else {
                sToT.put(s.charAt(i), t.charAt(i));
            }
            if (tToS.containsKey(t.charAt(i))) {
                if (tToS.get(t.charAt(i)) != s.charAt(i)) {
                    return false;
                }
            } else {
                tToS.put(t.charAt(i), s.charAt(i));
            }
        }
        return true;
    }
}